src/components/DateInput.jsx

```diff
@@ -109,11 +109,8 @@ export default class DateInput extends React.Component {
           'DateInput--disabled': disabled,
         })}
       >
-        <label className="DateInput__label" htmlFor={id}>
-          {placeholder}
-        </label>
-
         <input
+          aria-label={placeholder}
```

Comment:
note that this prop defaults to the empty string, so i think you might want `{placeholder || null}` here?